modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/components/OrderableTable.tsx

```diff
@@ -0,0 +1,322 @@
+
+			const targetSize = ref.current.getBoundingClientRect();
+			const targetCenter = (targetSize.bottom - targetSize.top) / 2;
+
+			const draggedOffset: {
+				x: number;
+				y: number;
+			} | null = monitor.getClientOffset();
+
+			if (!draggedOffset) {
+				return;
+			}
+
+			const draggedTop = draggedOffset.y - targetSize.top;
+
+			if (
+				(draggedIndex < targetIndex && draggedTop < targetCenter) ||
+				(draggedIndex > targetIndex && draggedTop > targetCenter)
+			) {
+				return;
+			}
+
+			onOrderChange({draggedIndex, targetIndex});
+
+			item.index = targetIndex;
+		},
+	});
+
+	dragRef(dropRef(ref));
+
+	return (
+		<ClayTable.Row
+			className={classNames('orderable-table-row', {
+				dragging: isDragging,
+			})}
+			key={index}
+			ref={ref}
+		>
+			<ClayTable.Cell className="drag-handle-cell">
+				<ClayButtonWithIcon
+					aria-label={Liferay.Util.sub(
+						Liferay.Language.get('drag-x'),
+						item.label || Liferay.Language.get('item')
+					)}
+					displayType={null}
+					symbol="drag"
+				/>
+			</ClayTable.Cell>
+
+			{fields.map((field) => {
+				const itemFieldValue = String(item[field.name]);
+
+				const fuzzyMatch = fuzzy.match(
+					query,
+					itemFieldValue,
+					FUZZY_OPTIONS
+				);
+
+				return (
+					<ClayTable.Cell key={field.name}>
+						{fuzzyMatch ? (
+							<span
+								dangerouslySetInnerHTML={{
+									__html: fuzzyMatch.rendered,
+								}}
+							/>
+						) : (
+							<span>{itemFieldValue}</span>
+						)}
+					</ClayTable.Cell>
+				);
+			})}
+		</ClayTable.Row>
+	);
+};
+
+interface OrderableTableInterface {
+	fields: Array<{
+		label: string;
+		name: string;
+	}>;
+	items: Array<any>;
+	noItemsButtonLabel: string;
+	noItemsDescription: string;
+	noItemsTitle: string;
+	onCancelButtonClick: Function;
+	onCreationButtonClick: Function;
+	onOrderChange: Function;
+	onSaveButtonClick: Function;
+	title: string;
+}
+
+const OrderableTable = ({
+	fields,
+	items: initialItems,
+	noItemsButtonLabel,
+	noItemsDescription,
+	noItemsTitle,
+	onCancelButtonClick,
+	onCreationButtonClick,
+	onOrderChange,
+	onSaveButtonClick,
+	title,
+}: OrderableTableInterface) => {
+	const [items, setItems] = useState(initialItems);
+	const [query, setQuery] = useState('');
+
+	const onSearch = (query: string) => {
+		setQuery(query);
+
+		const regexp = new RegExp(query, 'i');
+
+		setItems(
+			query
+				? initialItems.filter((item) =>
+						fields.some((field) =>
+							String(item[field.name]).match(regexp)
+						)
+				  ) || []
+				: initialItems
+		);
+	};
+
+	const handleOnOrderChange = ({
+		draggedIndex,
+		targetIndex,
+	}: {
+		draggedIndex: number;
+		targetIndex: number;
+	}) => {
+		const orderedItems = items.map((item, index) => {
+			if (index === draggedIndex) {
+				return items[targetIndex];
+			}
+
+			if (index === targetIndex) {
+				return items[draggedIndex];
+			}
+
+			return item;
+		});
+
+		setItems(orderedItems);
+
+		onOrderChange({
+			orderedItems,
+		});
```

Comment:
This is a bit of a smell to set the same value both internally and also through on `onChange` prop. Typically you would only have one or the other. In this case I think you would only have `onOrderChange` or just an `onChange` and remove the Internal state.